tests/acceptance/features/apiSharingNg/reshare.feature

```diff
@@ -0,0 +1,125 @@
+  So that they can have access to it
+
+  https://owncloud.dev/libre-graph-api/#/drives.permissions/Invite
+
+  Background:
+    Given these users have been created with default attributes and without skeleton files:
+      | username |
+      | Alice    |
+      | Brian    |
+      | Carol    |
+
+
+  Scenario Outline: reshare a file to a user with different roles
+    Given user "Alice" has uploaded file with content "to share" to "/textfile1.txt"
+    And user "Alice" has sent the following share invitation:
+      | resourceType    | file               |
+      | resource        | textfile1.txt      |
+      | space           | Personal           |
+      | sharee          | Brian              |
+      | shareType       | user               |
+      | permissionsRole | <permissions-role> |
+    When user "Brian" sends the following share invitation using the Graph API:
+      | resourceType    | file                       |
+      | resource        | textfile1.txt              |
+      | space           | Shares                     |
+      | sharee          | Carol                      |
+      | shareType       | user                       |
+      | permissionsRole | <reshare-permissions-role> |
+    Then the HTTP status code should be "200"
+    And for user "Carol" the space Shares should contain these entries:
+      | textfile1.txt |
+    Examples:
+      | permissions-role | reshare-permissions-role |
+      | Viewer           | Viewer                   |
+      | File Editor      | Viewer                   |
+      | File Editor      | File Editor              |
+
+
+  Scenario Outline: reshare a folder to a user with different roles
+    Given user "Alice" has created folder "FolderToShare"
+    And user "Alice" has sent the following share invitation:
+      | resourceType    | folder             |
+      | resource        | FolderToShare      |
+      | space           | Personal           |
+      | sharee          | Brian              |
+      | shareType       | user               |
+      | permissionsRole | <permissions-role> |
+    When user "Brian" sends the following share invitation using the Graph API:
+      | resourceType    | folder                     |
+      | resource        | FolderToShare              |
+      | space           | Shares                     |
+      | sharee          | Carol                      |
+      | shareType       | user                       |
+      | permissionsRole | <reshare-permissions-role> |
+    Then the HTTP status code should be "200"
+    And for user "Carol" the space Shares should contain these entries:
+      | FolderToShare |
+    Examples:
+      | permissions-role | reshare-permissions-role |
+      | Viewer           | Viewer                   |
+      | Editor           | Viewer                   |
+      | Editor           | Editor                   |
+      | Editor           | Uploader                 |
```

Comment:
can other combination be made?

Reply:
same thing as in comment https://github.com/owncloud/ocis/pull/8275#discussion_r1464417031